mitiq/zne/inference.py

```diff
@@ -920,6 +922,136 @@ def reduce(self) -> float:
         return self._zne_limit
 
 
+class RungeFactory(BatchedFactory):
+    """Factory object implementing another version of Richardson extrapolation.
+    In this version the original set of nodes (scale_factors) are mapped to set
+    of fake nodes known as Chebyshev-Lobatto points. This method gives a better
+    interpolation when there are a lot of nodes, especially close to the
+    edges of the domain.
+
+    Reference:
+        https://www.sciencedirect.com/science/article/abs/pii/S0377042719303449
+
+    Args:
+        scale_factors: Sequence of noise scale factors at which
+                       expectation values should be measured.
+        shot_list: Optional sequence of integers corresponding to the number
+                   of samples taken for each expectation value. If this
+                   argument is explicitly passed to the factory, it must have
+                   the same length of scale_factors and the executor function
+                   must accept "shots" as a valid keyword argument.
+
+    Raises:
+        ValueError: If data is not consistent with the extrapolation model.
+        ExtrapolationWarning: If the extrapolation fit is ill-conditioned.
+    """
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+        float,
+        Tuple[
+            float,
+            Union[float, None],
+            List[float],
+            Union[np.ndarray, None],
+            Callable[[float], float],
+        ],
+    ]:
+        # Richardson extrapolation is a particular case of a polynomial fit
+        # with order equal to the number of data points minus 1.
+        # In this version of Richardson extrapolation we map the samples to
+        # a new set of fake nodes and polynomially extrapolate.
+
+        order = len(scale_factors) - 1
+
+        # Define interval [a, b] for which the scale_factors are mapped to
+        a = 1.0
```

Comment:
```suggestion
        # Define interval [a, b] for which the scale_factors are mapped to
        a = min(scale_factors)
```
Not sure what is the most appropriate choice for this interval. I should check the details of the algorithm to get some intuition.  For example:
`a = min(scale_factors)` or even `a=0.0` could both be reasonable choices.

Reply:
I agree it should be `a = 0.0`, but testing it I found that `a = 1.0` generally gave better results. I am going to assume I somehow missed something, so I will change it to `a = min(scale_factors)` for now.

Reply:
It seems like the Lebesgue constant explodes since we are not sampling near zero. Check out theorem 2.4 and eq.9 in the paper.